ios/brave-ios/Sources/Brave/Helpers/URLBarHelper.swift

```diff
@@ -0,0 +1,183 @@
+// Copyright 2024 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import Foundation
+import UIKit
+
+class URLBarHelper {
+
+  static let shared = URLBarHelper()
+
+  func shouldShowSearchSuggestions(using lastReplacement: String) -> Bool {
+    // Check if last entry to url textfield needs to be checked as suspicious.
+    // The reason of checking count is bigger than 1 is the single character
+    // entries will always be safe and only way to achieve multi character entry is
+    // using paste board.
+    // This check also allow us to handle paste permission case
+    guard lastReplacement.count > 1 else {
+      return true
+    }
+
+    // Check if paste board has any text to guarantee the case
+    guard UIPasteboard.general.hasStrings || UIPasteboard.general.hasURLs else {
+      return true
+    }
+
+    // Perform check on pasted text
+    if let pasteboardContents = UIPasteboard.general.string {
+      return !isSuspiciousQuery(pasteboardContents)
+    }
+
+    return true
+  }
+
+  /// Whether  the desired text should allow search suggestions to appear when it is copied
+  /// - Parameter query: Search query copied
+  /// - Returns: the result if it is suspicious
+  func isSuspiciousQuery(_ query: String) -> Bool {
+    // Remove the msg if the query is too long
+    if query.count > 50 {
+      return true
+    }
+
+    // Remove the msg if the query contains more than 7 words
+    if query.components(separatedBy: " ").count > 7 {
+      return true
+    }
+
+    // Remove the msg if the query contains a number longer than 7 digits
+    if let _ = checkForLongNumber(query, 7) {
+      return true
+    }
+
+    // Remove if email (exact), even if not totally well formed
+    if checkForEmail(query) {
+      return true
+    }
+
+    // Remove if query looks like an http pass
+    if query.range(of: "[^:]+:[^@]+@", options: .regularExpression) != nil {
```

Comment:
don't we have better parsers to check for urls instead of regexing? e.g GURL

Reply:
The plan was to rewrite this function from BraveSearch so the same logic and regexes are used. Same for the unit tests, it is rewritten 1:1 from the corresponding BraveSearch backend class

Reply:
yeah basically explained in PR description and this comment

https://github.com/brave/brave-core/pull/24376#discussion_r1657350574

It is 1:1 implementation of isSuspiciousQuery from brave search and this is not doing any edits related with Url display and such. Just deciding of copied text should show search suggestions or not.